include/libopencm3/stm32/l0/usart.h

```diff
@@ -0,0 +1,321 @@
+
+#define USART1				USART1_BASE
+#define USART2				USART2_BASE
+#define LPUART1				LPUART1_BASE
+
+/*****************************************************************************/
+/* Register definitions                                                      */
+/*****************************************************************************/
+
+#define USART_CR1(usart_base)		MMIO32(usart_base + 0x00)
+#define USART1_CR1			USART_CR1(USART1_BASE)
+#define USART2_CR1			USART_CR1(USART2_BASE)
+
+#define USART_CR2(usart_base)		MMIO32(usart_base + 0x04)
+#define USART1_CR2			USART_CR2(USART1_BASE)
+#define USART2_CR2			USART_CR2(USART2_BASE)
+
+#define USART_CR3(usart_base)		MMIO32(usart_base + 0x08)
+#define USART1_CR3			USART_CR3(USART1_BASE)
+#define USART2_CR3			USART_CR3(USART2_BASE)
+
+#define USART_BRR(usart_base)		MMIO32(usart_base + 0x0c)
+#define USART1_BRR			USART_BRR(USART1_BASE)
+#define USART2_BRR			USART_BRR(USART2_BASE)
+
+#define USART_GTPR(usart_base)		MMIO32(usart_base + 0x10)
+#define USART1_GTPR			USART_GTPR(USART1_BASE)
+#define USART2_GTPR			USART_GTPR(USART2_BASE)
+
+#define USART_RTOR(usart_base)		MMIO32(usart_base + 0x14)
+#define USART1_RTOR			USART_RTOR(USART1_BASE)
+#define USART2_RTOR			USART_RTOR(USART2_BASE)
+
+#define USART_RQR(usart_base)		MMIO32(usart_base + 0x18)
+#define USART1_RQR			USART_RQR(USART1_BASE)
+#define USART2_RQR			USART_RQR(USART2_BASE)
+
+#define USART_ISR(usart_base)		MMIO32(usart_base + 0x1c)
+#define USART1_ISR			USART_ISR(USART1_BASE)
+#define USART2_ISR			USART_ISR(USART2_BASE)
+
+#define USART_ICR(usart_base)		MMIO32(usart_base + 0x20)
+#define USART1_ICR			USART_ICR(USART1_BASE)
+#define USART2_ICR			USART_ICR(USART2_BASE)
+
+#define USART_RDR(usart_base)		MMIO8(usart_base + 0x24)
+#define USART1_RDR			USART_RDR(USART1_BASE)
+#define USART2_RDR			USART_RDR(USART2_BASE)
+
+#define USART_TDR(usart_base)		MMIO8(usart_base + 0x28)
+#define USART1_TDR			USART_TDR(USART1_BASE)
+#define USART2_TDR			USART_TDR(USART2_BASE)
+
+/*****************************************************************************/
+/* Register values                                                           */
+/*****************************************************************************/
+
+/* USART_CR1 Values ---------------------------------------------------------*/
+
+#define USART_CR1_M1			(1 << 28) /* F07x */
+#define USART_CR1_EOBIE			(1 << 27)
+#define USART_CR1_RTOIE			(1 << 26)
+
+#define USART_CR1_DEAT_SHIFT		21
+#define USART_CR1_DEAT			(0x1F << USART_CR1_DEAT_SHIFT)
+#define USART_CR1_DEAT_VAL(x)		((x) << USART_CR1_DEAT_SHIFT)
+
+#define USART_CR1_DEDT_SHIFT		16
+#define USART_CR1_DEDT			(0x1F << USART_CR1_DEDT_SHIFT)
+#define USART_CR1_DEDT_VAL(x)		((x) << USART_CR1_DEDT_SHIFT)
+
+#define USART_CR1_OVER8			(1 << 15)
+#define USART_CR1_CMIE			(1 << 14)
+#define USART_CR1_MME			(1 << 13)
+#define USART_CR1_M			(1 << 12) /* Obsolete, please use M0 */
```

Comment:
If it's obsolete, why would you add it in a new file?  If this is copy pasta from the f0 file, then it's time to pull up the common code.  you're going to see lots of the same code in the f3 uart too.